File: server/app/model/database/game-constants.ts
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { ApiProperty } from '@nestjs/swagger';
import { Document } from 'mongoose';

export type GameConstantsDocument = GameConstants & Document;

@Schema()
export class GameConstants {
    @ApiProperty()
    @Prop({ required: true })
    gameDuration: number;

    @ApiProperty()
    @Prop({ required: true })
    penaltyTime: number;

    @ApiProperty()
    @Prop({ required: true })
    bonusTime: number;

    @ApiProperty()
    _id?: string;
}

export const gameConstantsSchema = SchemaFactory.createForClass(GameConstants);

File: android-client/lib/classes/game_constants.dart
class GameConstants {
  int gameDuration;
  int penaltyTime;
  int bonusTime;
  bool cheatMode;

  GameConstants({required this.gameDuration, required this.penaltyTime, required this.bonusTime, required this.cheatMode});

  factory GameConstants.fromJson(Map<String, dynamic> json) {
    return GameConstants(
      gameDuration: json['gameDuration'] as int,
      penaltyTime: json['penaltyTime'] as int,
      bonusTime: json['bonusTime'] as int,
      cheatMode: json['cheatMode'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'gameDuration': gameDuration,
      'penaltyTime': penaltyTime,
      'bonusTime': bonusTime,
      'cheatMode': cheatMode,
    };
  }
}
